### PR TITLE
Issues/22 add spacewalk pallet

### PR DESCRIPTION
**Pull request by xiuxiu62**
Add and configure orml and spacewalk pallets

### CODE REVIEW

**Comment by ebma** [comment]
I checked out the PR but when I `cargo build --release` I am running into
```
error[E0277]: the trait bound `H256: TypeInfo` is not satisfied
   --> /home/marcel/.cargo/git/checkouts/substrate-7e08433d4c370a21/fc3fd07/primitives/runtime/src/testing.rs:260:2
    |
260 |     type Hash = <Header as traits::Header>::Hash;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TypeInfo` is not implemented for `H256`
    |
note: required by a bound in `traits::Block::Hash`
   --> /home/marcel/.cargo/git/checkouts/substrate-7e08433d4c370a21/fc3fd07/primitives/runtime/src/traits.rs:701:5
    |
688 |     type Hash: Member
    |          ---- required by a bound in this
...
701 |         + TypeInfo;
    |           ^^^^^^^^ required by this bound in `traits::Block::Hash`
...
error: could not compile `sp-runtime` due to 3 previous errors
```
I'm running `cargo 1.58.0-nightly (2e2a16e98 2021-11-08)` since the repository contains a `rust-toolchain` file.
@xiuxiu62 Any ideas on what the issue here might be?

**Comment by xiuxiu62** [comment]
interesting, it's compiling for me.  I tried clearing my registry too, but can't reproduce this.  You might be able to try a cargo update or clearing ~/.cargo/registry

**Comment by xiuxiu62** [comment]
> I checked out the PR but when I `cargo build --release` I am running into
> 
> ```
> error[E0277]: the trait bound `H256: TypeInfo` is not satisfied
>    --> /home/marcel/.cargo/git/checkouts/substrate-7e08433d4c370a21/fc3fd07/primitives/runtime/src/testing.rs:260:2
>     |
> 260 |     type Hash = <Header as traits::Header>::Hash;
>     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TypeInfo` is not implemented for `H256`
>     |
> note: required by a bound in `traits::Block::Hash`
>    --> /home/marcel/.cargo/git/checkouts/substrate-7e08433d4c370a21/fc3fd07/primitives/runtime/src/traits.rs:701:5
>     |
> 688 |     type Hash: Member
>     |          ---- required by a bound in this
> ...
> 701 |         + TypeInfo;
>     |           ^^^^^^^^ required by this bound in `traits::Block::Hash`
> ...
> error: could not compile `sp-runtime` due to 3 previous errors
> ```
> 
> I'm running `cargo 1.58.0-nightly (2e2a16e98 2021-11-08)` since the repository contains a `rust-toolchain` file. @xiuxiu62 Any ideas on what the issue here might be?

Ah wait I'm helping Nemanja fix the spacewalk client so we can integrate it into pendulum and this is the error message I'm getting, compiling the clients.  Did you try compiling that instead of this by accident?

**Comment by ebma** [comment]
Yeah the error message also reminded me a lot of that but I'm pretty sure I checked out this branch and tried to compile it haha. I don't know, maybe I didn't `cargo clean` and that's why. But `cargo build --release` is also working for me now.

**Review by ebma — APPROVED** [review]
Looks good to me. I also tested if the extrinsic calls of the pallet work and they do.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -461,8 +522,11 @@ construct_runtime!(
 		ParachainInfo: parachain_info::{Pallet, Storage, Config} = 3,
 
 		// Monetary stuff.
-		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>} = 10,
-		TransactionPayment: pallet_transaction_payment::{Pallet, Storage} = 11,
+		Tokens: orml_tokens::{Pallet, Call, Storage, Event<T>} = 10,
+		Currencies: orml_currencies::{Pallet, Call, Storage, Event<T>} = 11,
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>} = 12,
+		TransactionPayment: pallet_transaction_payment::{Pallet, Storage} = 13,
+		Spacewalk: pallet_spacewalk::{Pallet, Call, Storage, Event<T>} = 14,
```
[the remark this answers is not in the document]
Comment (xiuxiu62) [reply runtime/src/lib.rs:529]: oh not particularly, this just looked nice for some reason haha. I can certainly change it

**Pull request merged**